src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,83 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React from 'react';
+import { TodoStatus } from '../../types/TodoStatus ';
+
+type Props = {
+  onChangeQuery: (query: string) => void,
+  onChangeSelect: (event: string) => void,
+  onSetSelectedOption: React.Dispatch<React.SetStateAction<string>>
+  query: string,
+  selectedOption: string,
+
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onChangeQuery = () => {},
+  onChangeSelect = () => {},
+  query,
+  selectedOption,
+  onSetSelectedOption,
+}) => {
+  const onChangeSetQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChangeQuery(event.target.value);
+  };
+
+  const removeQuery = () => onChangeQuery('');
+
+  const handleChangeOption
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onSetSelectedOption(event.target.value);
+    onChangeSelect(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span
+          className="select"
+        >
+          <select
+            data-cy="statusSelect"
+            id="select"
+            value={selectedOption}
+            onChange={handleChangeOption}
+          >
+            <option value={TodoStatus.All}>All</option>
+            <option value={TodoStatus.Active}>Active</option>
+            <option value={TodoStatus.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p
+        className="control is-expanded has-icons-left has-icons-right"
+      >
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={onChangeSetQuery}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query && (
+          <span
+            className="icon is-right"
+            style={{ pointerEvents: TodoStatus.All }}
```

Comment:
You don't need to use `enum` here
Even better would be to move this style to some `class` styles